CorsixTH/Lua/entities/patient.lua

```diff
@@ -879,7 +879,7 @@ function Patient:tickDay()
             name = current.name,
             x = current.x,
             y = current.y,
-            must_happen = current.must_happen,
+            must_happen = current.name == "walk" and current.saved_must_happen or current.name == "idle" and current.must_happen,
```

Comment:
Please make this clearer with brackets, it's too easy to forget precedence of `and` and `or`. As is it's a bit hard to read.